framework/Optimizers/stepManipulators/GradientHistory.py

```diff
@@ -0,0 +1,300 @@
+from __future__ import division, print_function, unicode_literals, absolute_import
+#End compatibility block for Python 3----------------------------------------------------------------
+
+#External Modules------------------------------------------------------------------------------------
+import abc
+import numpy as np
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from utils import InputData, InputTypes, mathUtils, randomUtils
+from .StepManipulator import StepManipulator
+from . import NoConstraintResolutionFound
+#Internal Modules End--------------------------------------------------------------------------------
+
+class GradientHistory(StepManipulator):
+  """
+    Changes step size depending on history of gradients
+  """
+  requiredInformation = ['gradientHist', 'prevStepSize']
+  optionalInformation = ['recommend']
+
+  ##########################
+  # Initialization Methods #
+  ##########################
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for specifying input of cls.
+    """
+    specs = super(GradientHistory, cls).getInputSpecification()
+    specs.description = r"""if this node is present, indicates that the iterative steps in the gradient
+        descent algorithm should be determined by the sequential change in gradient. In particular, rather
+        than using the magnitude of the gradient to determine step size, the directional change of the
+        gradient versor determines whether to take larger or smaller steps. If the gradient in two successive
+        steps changes direction, the step size shrinks. If the gradient instead continues in the same
+        direction, the step size grows. The rate of shrink and growth are controlled by the \xmlNode{shrinkFactor}
+        and \xmlNode{growthFactor}. Note these values have a large impact on the optimization path taken.
+        Large growth factors converge slowly but explore more of the input space; large shrink factors
+        converge quickly but might converge before arriving at a local minimum."""
+    specs.addSub(InputData.parameterInputFactory('growthFactor', contentType=InputTypes.FloatType,
+        descr=r"""specifies the rate at which the step size should grow if the gradient continues in
+              same direction through multiple iterative steps. For example, a growth factor of 2 means
+              that if the gradient is identical twice, the step size is doubled. \default{1.25} """))
+    specs.addSub(InputData.parameterInputFactory('shrinkFactor', contentType=InputTypes.FloatType,
```

Comment:
@joshua-cogliati-inl we really need a way to specify default values in "createClass" if nodes and or params are optional